addons/web/static/src/js/widgets/domain_selector.js

```diff
@@ -652,6 +652,7 @@ var DomainLeaf = DomainNode.extend({
             this.chain !== undefined ? this.chain.toString().split(".") : [],
             this.options
         );
+
```

Comment:
useless diff

Reply:
not sure about this, but I have a small question: you know that all fields are available in the search view (in group by menu and in advanced filter mode), so, is that a problem in your case?

Otherwise, the proposal looks all right I guess. I would use the word `hidden` or `technical` instead of `exclude`, and you should consider adding a test as well.

Reply:
well, it could stay in search view as we already have an error message if we are not in the context of a mailing list.
In fact, the field MUST remains in the search view if we are in that context of a mailing list.
cc: @dja-odoo @pka-odoo and see comment above

Reply:
@dbeguin @ged-odoo Well this is not exactly what I had in mind, because this is not very common case. I would keep the Py code intact, instead, I would add a node option (in xml on `domain` widget that can contain the list of fields we don't want to display in field selector). Do let me know your opinion.

Reply:
Would be more precise in naming, like view_hidden or view_exclude, as this is just a web-client property.

Otherwise, we need to be able to filter out opt-out contacts in a mailing list context, because that's the core use of this field. If it is not displayed in this view this PR is probably not acceptable. Indeed using mailing list and opt-out contacts is core behavior of mass mailing.